### PR TITLE
stm32 dmamux fix function array definitions

### PR DESCRIPTION
**Pull request by str4t0m**
This commit uses dt property dma-channels instead of ll defines to
make sure every soc has the correct number of function pointers defined.

While commit 5f6218a fixed this for g0 series, it caused a regression for
other series(e.g. stm32g431). Using UTIL_LISTIFY and dt properties
this should finally be fixed while also reducing boilerplate code.

These variables currently are not used anywhere in the driver, so we could consider removing them completely.
Additionally I believe defining arrays of function pointers is quite wasteful in case the functions only reset or query the value of a single register. Defining macros to access these registers directly would spare >100 Bytes of flash on a soc with 12 channels.
Should I change the pr to define a few macros to access these registers instead of defining the function pointer arrays?

### CODE REVIEW

File: drivers/dma/dmamux_stm32.c [hunk 1]
```diff
@@ -45,6 +52,36 @@ struct dmamux_stm32_config {
...
+#define IS_ACTIVE_FLAG_SOX(i, _)	LL_DMAMUX_IsActiveFlag_SO  ## i,
+#define CLEAR_FLAG_SOX(i, _)		LL_DMAMUX_ClearFlag_SO ## i,
+#define IS_ACTIVE_FLAG_RGOX(i, _)	LL_DMAMUX_IsActiveFlag_RGO  ## i,
+#define CLEAR_FLAG_RGOX(i, _)		LL_DMAMUX_ClearFlag_RGO ## i,
+
+uint32_t table_ll_channel[] = {
+	UTIL_LISTIFY(DT_INST_PROP(0, dma_channels), DMAMUX_CHANNEL)
+};
+
+uint32_t (*func_ll_is_active_so[])(DMAMUX_Channel_TypeDef * DMAMUXx) = {
```
Comment (str4t0m) [inline drivers/dma/dmamux_stm32.c:69]: As raised in another pr, the pointer asterisk should be next to the name.
But checkpatch seems to interpret this as multiplication(ctx:WxV)  so I had to change it here to comply with the check.
Grepping for `\*\ const` shows about 500 occurrences, so am I right in assuming that this not so strict? How should such false positives be handled?

Comment (erwango) [reply drivers/dma/dmamux_stm32.c:69]: For false positive, you should raise the point to merging people, so the PR is merged despite failing CI.

**Review by FRASTM — APPROVED** [review]

**Comment by str4t0m** [comment]
The compliance check is a false positive.
`DMAMUX_Channel_TypeDef *DMAMUXx` is interpreted as a multiplication.

**Comment by galak** [comment]
> The compliance check failure is a false positive.
> `DMAMUX_Channel_TypeDef *DMAMUXx` is interpreted as a multiplication.

 I don't think this is a false positive.  Its the coding style to have a space between the `*` so this should be:
`DMAMUX_Channel_TypeDef * DMAMUXx`

**Comment by str4t0m** [comment]
May I ask what takes precedence over the simple 

> preferred use of * is adjacent to the data name or function name

 rule in this case to be able to regard that in the future?

**Comment by galak** [comment]
> May I ask what takes precedence over the simple
> 
> > preferred use of * is adjacent to the data name or function name
> 
> rule in this case to be able to regard that in the future?

I think you are correct about this being a false positive on checkpatch.

File: drivers/dma/dmamux_stm32.c [hunk 2]
```diff
@@ -45,6 +52,36 @@ struct dmamux_stm32_config {
 	const struct dmamux_stm32_channel *mux_channels;
 };
 
+/*
+ * UTIL_LISTY is used to generte arrays with function pointers to check
```
Comment (galak) [inline drivers/dma/dmamux_stm32.c:56]: generte -> generate

Comment (galak) [reply drivers/dma/dmamux_stm32.c:56]: UTIL_LISTY -> UTIL_LISTIFY

Comment (str4t0m) [reply drivers/dma/dmamux_stm32.c]: Fixed the typos now.

**Review by erwango — APPROVED** [review]

**Pull request merged**